### PR TITLE
Feature/hir 124 api for manual merge accommodations data

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

File: HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs [hunk 1]
```diff
@@ -120,6 +120,19 @@ public async Task<IActionResult> AddManualCorrectionToAccommodation(int accommod
         }
 
 
+        [HttpPost("accommodations/uncertainMatches/{uncertainMatchId}/match")]
```
Comment (kirillta) [inline HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs:123]: `uncertain-matches` at least, but better `accommodations/matches/{uncertainMatchId}/uncertain`

File: HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs [hunk 2]
```diff
@@ -20,6 +20,37 @@ public AccommodationService(NakijinContext context, IAccommodationsDataMerger ac
...
 
 
+        public async Task<Result> MatchUncertain(int uncertainMatchId)
+        {
+            var uncertainMatch = await _context.AccommodationUncertainMatches.SingleOrDefaultAsync(um => um.Id == uncertainMatchId);
+
+            if (uncertainMatch == default)
+                return Result.Failure($"Uncertain match with id {uncertainMatchId} does not exist.");
+
+            if (!uncertainMatch.IsActive)
```
Comment (kirillta) [inline HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs:30]: Could this be merged with the upper statement? Like 
```csharp
await _context.AccommodationUncertainMatches
    .Where(um => um.IsActive)
    .SingleOrDefaultAsync(um => um.Id == uncertainMatchId);
```

File: HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs [hunk 3]
```diff
@@ -20,6 +20,37 @@ public AccommodationService(NakijinContext context, IAccommodationsDataMerger ac
...
+            var secondAccommodation = await _context.Accommodations.SingleOrDefaultAsync(ac => ac.Id == uncertainMatch.NewHtId);
+
+            foreach (var supplierAccommodation in secondAccommodation.SupplierAccommodationCodes)
+                firstAccommodation.SupplierAccommodationCodes.TryAdd(supplierAccommodation.Key, supplierAccommodation.Value);
+
+            firstAccommodation.IsCalculated = false;
+            secondAccommodation.IsActive = false;
+            uncertainMatch.IsActive = false;
+
+            _context.Update(firstAccommodation);
```
Comment (kirillta) [inline HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs:43]: Looks like a good place for transaction

**Review by kirillta — CHANGES_REQUESTED** [review]

File: HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs (hunk 2 above)
Comment (hasmikharutyunyan) [reply HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs:30]: Yes, but  we decided to not throw error in that kind of situations.

File: HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs (hunk 3 above)
Comment (hasmikharutyunyan) [reply HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs:43]: Yes, `SaveChanges` by default opened an transaction (here is one `Savechanges`)

File: HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs (hunk 1 above)
Comment (hasmikharutyunyan) [reply HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs:123]: Here `match` is used as verb, we match that was uncertain matched.

Comment (kirillta) [reply HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs]: Then `accommodations/{uncertainMatchId}/match-uncertain`?

Comment (hasmikharutyunyan) [reply HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs]: I understand what You want, but `uncertainmatchId`  is not an accommodation Id, and here will be not understandable I think

File: HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs (hunk 2 above)
Comment (kirillta) [reply HappyTravel.PropertyManagement.Api/Services/AccommodationService.cs:30]: As for me both conditions are safe to return success

File: HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs (hunk 1 above)
Comment (kirillta) [reply HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs]: Agree

File: HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs [hunk 4]
```diff
@@ -120,6 +120,19 @@ public async Task<IActionResult> AddManualCorrectionToAccommodation(int accommod
         }
```
Comment (oleg1std) [inline HappyTravel.PropertyManagement.Api/Controllers/AccommodationsController.cs:121]: The controller contains get accommodations methods with incorrect response type in [ProducesResponseType]

**Review by oleg1std — APPROVED** [review]

**Review by kirillta — APPROVED** [review]

**Pull request merged**